### PR TITLE
[proofs] [doc] Minor changes to general proofs page, some changes to Alethe page, added DOT page

### PR DESCRIPTION
**Pull request by HanielB**
…Alethe page

### CODE REVIEW

**Comment by Lachnitt** [comment]
The Alethe part looks good to me!

File: docs/proofs/proofs.rst [hunk 1]
```diff
@@ -1,14 +1,15 @@
-Proof generation
+Proof production
 ================
 
-cvc5 can generate formal proofs and output it in the following formats:
+cvc5 produces proofs in an internal proof reflecting faithfully reflecting its
+reasoning. Here is a comprehensive description of the :doc:`proof rules
+<proof_rules>`.
```
[the remark this answers is not in the document]
Comment (nafur) [reply docs/proofs/proofs.rst:6]: Yes, it does. (see https://cvc5.github.io/docs-ci/docs-pr8501/proofs/proofs.html)

File: docs/proofs/proofs.rst [hunk 2]
```diff
@@ -1,14 +1,15 @@
-Proof generation
+Proof production
 ================
 
-cvc5 can generate formal proofs and output it in the following formats:
+cvc5 produces proofs in an internal proof reflecting faithfully reflecting its
+reasoning. Here is a comprehensive description of the :doc:`proof rules
```
Comment (nafur) [inline docs/proofs/proofs.rst:5]:
```suggestion
cvc5 produces proofs in an internal proof calculus faithfully reflecting its
reasoning. Here is a comprehensive description of the :doc:`proof rules
```
I guess...?

Comment (ajreynol) [reply docs/proofs/proofs.rst:5]: I modified your comment slightly.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**